docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+
+#### Notes
+
+These are usually `"application"` type subnets and serve as a fallback option.
+
+
+
+## `total_cycles_minted`
+
+Returns the total number of cycles ever minted by the CMC since its genesis. This provides a measure of the overall cycles created via the CMC mechanism.
+
+#### Parameters
+
+```candid
+record {}
+```
+
+#### Returns
+
+```candid
+nat // Total cycles minted as an arbitrary-precision natural number.
+```
+
+
+
+## `get_build_metadata`
+
+Returns metadata about the current build of the Cycles Minting Canister.
+
+#### Parameters
+
+```candid
+record {}
+```
+
+#### Returns
+
+```candid
+// Current type from define_get_build_metadata_candid_method!: text
+// Draft type shown for simplicity:
+record {
+  repo: opt text;
+  git_revision: opt text;
+  rust_version: opt text;
+  cargo_version: opt text;
+  build_time: opt text;
+}
+```
+
+-   `repo`: URL of the source code repository.
+-   `git_revision`: Git commit hash.
+-   `rust_version`: Version of Rust used.
+-   `cargo_version`: Version of Cargo used.
+-   `build_time`: UTC string of when the build was performed.
+
+#### Notes
+
+This is primarily used for debugging, reproducibility, and audits. Fields may be `null` depending on build environment (though typically returned as a single formatted string).
+
+
+
+## `http_request` (Metrics endpoint)
+
+Standard HTTP interface for fetching canister metrics in Prometheus format.
+
+#### Parameters
+
+```candid
+record {
+  url: text;
+  method: text; // Should be "GET"
+  headers: vec record { name: text; value: text };
+  body: blob;
+}
+```
+
+#### Returns
+
+```candid
+record {
+  status_code: nat16;
+  headers: vec record { name: text; value: text };
+  body: blob; // Prometheus metrics data
+  // streaming_strategy : opt ... // Optional streaming fields omitted
+}
+```
+
+#### Notes
+
+-   Call with method "GET" and URL "/metrics".
+-   Provides metrics like `cmc_cycles_minted_total`, `cmc_icp_xdr_conversion_rate`, `cmc_limiter_cycles`, `cmc_cycles_limit`, etc. Useful for monitoring CMC activity and capacity. Requires standard HTTP request structure.
+
+
+
+## Shared logic
+
+### `CanisterSettings`
+
+The `CanisterSettings` record is used to configure the canister at the time of creation or through governance. Each field is optional and will default to system-provided values if not specified.
+
+```candid
+record {
+  controllers : opt vec principal;         // List of principals allowed to control the canister
+  compute_allocation : opt nat;            // Percentage (0-100) of guaranteed compute capacity
+  memory_allocation : opt nat;             // Memory reserved for the canister, in bytes
+  freezing_threshold : opt nat;            // Number of seconds the canister can go without being topped up before being frozen
+  // reserved_cycles_limit : opt nat;      // (Note: This field might not be directly settable via CMC)
```

Comment:
I don't understand the note here. 
All of this cannot be set by CMC but needs to be set by its controller (NNS root), right?